plugwise/smile.py

```diff
@@ -905,25 +900,31 @@ def get_power_data_from_location(self, loc_id):
                     key_string = f"{measurement}_{peak}_{log_found}"
                     net_string = f"net_electricity_{log_found}"
                     val = loc_logs.find(locator).text
+                    if attrs[ATTR_UNIT_OF_MEASUREMENT] == POWER_WATT:
+                        f_val = int(round(float(val)))
+                    else:
+                        f_val = format_measure(val, attrs[ATTR_UNIT_OF_MEASUREMENT])
                     if all(
                         item in key_string for item in ["electricity", "cumulative"]
                     ):
                         f_val = format_measure(val, ENERGY_KILO_WATT_HOUR)
-                    else:
-                        f_val = format_measure(val, attrs[ATTR_UNIT_OF_MEASUREMENT])
-                    if "gas" in measurement:
-                        key_string = f"{measurement}_{log_found}"
-                        f_val = float(f"{round(float(val), 3):.3f}")
-
                     # Energy differential
                     if "electricity" in measurement:
-                        f_val = float(f"{round(float(val), 1):.1f}")
                         diff = 1
                         if "produced" in measurement:
                             diff = -1
                         if net_string not in direct_data:
-                            direct_data[net_string] = float()
-                        direct_data[net_string] += float(f_val * diff)
+                            direct_data[net_string] = 0
+                        if f_val != 0:
+                            if isinstance(f_val, int):
```

Comment:
Same comment (just set it and overwrite with the if)

Reply:
And this one.

Reply:
Tried this, does not work due to the use of `+=`: the value is not replaced but added.

Reply:
attempt two: you are correct